File: 2020/day2.2.js
const fs = require('fs');

let results = [];

fs.readFile('./files/day2.txt', (err, data) => {

    if (err) {
        throw err;
    }
    
    const lines = data.toString().split('\n');
    for (let i = 0; i < lines.length; i++) {
        let count = 0;
        const policyValue = lines[i].split(':');
        const numbersLetters = policyValue[0].split(' ');
        const numbers = numbersLetters[0].split('-');

        policyValue[1] = policyValue[1].trim();
        
        const a = policyValue[1][numbers[0] - 1];
        const b = numbersLetters[1];

        const c = policyValue[1][numbers[1] - 1];

        if (a == b) {
            count++;
        }

        if (c == b) {
            count++;
        }

        if (count == 1) {
            results.push(i);
        }
    }

    
    console.log(results.length);
})

File: 2022/2022/day2/day2.2.swift
//
//  day2.2.swift
//  2022
//
//  Created by Teboho Mphure on 2022/12/04.
//

import Foundation

class Day2_2 : Day2_1 {
    var outcome = ["X": 0, "Y": 3, "Z": 6]
    func GetAmendedStrategyValue() {
        let lines = FileUtils.ReadFile(directory: "repos/advent-of-code/2022/2022/files/day2.txt")
        
        
        for line in lines {
            let pair = line.components(separatedBy: " ")
            
            if pair.count == 2 {
                var answer = 0
                let endResult: Int = outcome[pair[1]]!
                
                for key in myPlays.keys {
                    answer = EvaluateOptions(first: pair[0], second: key)
                    if endResult == answer {
                        total += answer + myPlays[key]!
                        break
                    }
                }
            }
        }
        
        print(total)
    }
}
